Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -561,7 +563,135 @@ void LeptonRecoSFProducer::Produce(ZJetEvent const& event,
+                return std::make_tuple(0.989, 0.001);
+            else if (200.0 < lepton.p4.P() && lepton.p4.P() < 300.0)
+                return std::make_tuple(0.986, 0.001);
+            else if (300.0 < lepton.p4.P() && lepton.p4.P() < 400.0)
+                return std::make_tuple(0.989, 0.001);
+            else if (400.0 < lepton.p4.P() && lepton.p4.P() < 600.0)
+                return std::make_tuple(0.983, 0.003);
+            else if (600.0 < lepton.p4.P() && lepton.p4.P() < 1500.0)
+                return std::make_tuple(0.986, 0.006);
+            else if (1500.0 < lepton.p4.P() && lepton.p4.P() < 3500.0)
+                return std::make_tuple(1.01, 0.01);
+        }
+    } else if (m_year=="2016") {
+        // https://twiki.cern.ch/twiki/bin/viewauth/CMS/MuonUL2016#High_pT_above_120_GeV
+        if (eta < 1.6) {
+            if (50.0 < lepton.p4.P() && lepton.p4.P() < 100.0)
+                return std::make_tuple(0.9914, 0.0008);
+            if (100.0 < lepton.p4.P() && lepton.p4.P() < 150.0)
+                return std::make_tuple(0.9936, 0.0009);
+            else if (150.0 < lepton.p4.P() && lepton.p4.P() < 200.0)
+                return std::make_tuple(0.993, 0.001);
+            else if (200.0 < lepton.p4.P() && lepton.p4.P() < 300.0)
+                return std::make_tuple(0.993, 0.002);
+            else if (300.0 < lepton.p4.P() && lepton.p4.P() < 400.0)
+                return std::make_tuple(0.990, 0.004);
+            else if (400.0 < lepton.p4.P() && lepton.p4.P() < 600.0)
+                return std::make_tuple(0.990, 0.003);
+            else if (600.0 < lepton.p4.P() && lepton.p4.P() < 1500.0)
+                return std::make_tuple(0.989, 0.004);
+            else if (1500.0 < lepton.p4.P() && lepton.p4.P() < 3500.0)
+                return std::make_tuple(0.8, 0.3);
+        }
+        else if (1.6 < eta && eta < 2.4) {
+            if (100.0 < lepton.p4.P() && lepton.p4.P() < 150.0)
+                return std::make_tuple(0.993, 0.001);
+            else if (150.0 < lepton.p4.P() && lepton.p4.P() < 200.0)
+                return std::make_tuple(0.991, 0.001);
+            else if (200.0 < lepton.p4.P() && lepton.p4.P() < 300.0)
+                return std::make_tuple(0.985, 0.001);
+            else if (300.0 < lepton.p4.P() && lepton.p4.P() < 400.0)
+                return std::make_tuple(0.981, 0.002);
+            else if (400.0 < lepton.p4.P() && lepton.p4.P() < 600.0)
+                return std::make_tuple(0.979, 0.004);
+            else if (600.0 < lepton.p4.P() && lepton.p4.P() < 1500.0)
+                return std::make_tuple(0.978, 0.005);
+            else if (1500.0 < lepton.p4.P() && lepton.p4.P() < 3500.0)
+                return std::make_tuple(0.9, 0.2);
+        }
+    } else if (m_year != "none") {
+        LOG(FATAL) << "No high pt RecoSF available for year " << m_year;
+    }
+    LOG(ERROR) << "No RecoSF available for this p and eta region. Using 1+/-0";
```

Comment:
This will not be reached as intended, since the year is checked first. You need to add this inside each year case, when you want to keep the implementation like this.

Reply:
Thanks. That's correct.

Reply:
Fixed.